src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpOverallOutcomeController.cs

```diff
@@ -0,0 +1,82 @@
+    public class RoatpOverallOutcomeController : Controller
+    {
+        private readonly IOutcomeApiClient _apiClient;
+        private readonly IApplicationApiClient _applicationApiClient;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IOverallOutcomeService _overallOutcomeService;
+        private readonly ILogger<RoatpOverallOutcomeController> _logger;
+
+        public RoatpOverallOutcomeController(IOutcomeApiClient apiClient, IQnaApiClient qnaApiClient,
+            IOverallOutcomeService overallOutcomeService, IApplicationApiClient applicationApiClient,
+            ILogger<RoatpOverallOutcomeController> logger)
+        {
+            _apiClient = apiClient;
+            _qnaApiClient = qnaApiClient;
+            _overallOutcomeService = overallOutcomeService;
+            _logger = logger;
+            _applicationApiClient = applicationApiClient;
+        }
+
+        [HttpGet]
+        [Authorize(Policy = "AccessApplication")]
+        public async Task<IActionResult> ProcessApplicationStatus(Guid applicationId)
+        {
+            var application = await _applicationApiClient.GetApplication(applicationId);
+            var model = _overallOutcomeService.BuildApplicationSummaryViewModel(application, User.GetEmail());
+
+            switch (application.ApplicationStatus)
+            {
+                case ApplicationStatus.New:
+                case ApplicationStatus.InProgress:
+                    return RedirectToAction("TaskList", "RoatpApplication", new {applicationId});
+                case ApplicationStatus.Successful:
+                {
+                    var oversightReview = await _apiClient.GetOversightReview(applicationId);
+                    if (oversightReview?.Status == OversightReviewStatus.SuccessfulAlreadyActive)
+                        return View("~/Views/Roatp/ApplicationApprovedAlreadyActive.cshtml", model);
+
+                    return View("~/Views/Roatp/ApplicationApproved.cshtml", model);
+                }
+                case ApplicationStatus.Unsuccessful: 
+                    if (application.GatewayReviewStatus == GatewayReviewStatus.Fail)
+                        return View("~/Views/Roatp/ApplicationUnsuccessful.cshtml", model);
+
+                    var unsuccessfulModel =
+                        await _overallOutcomeService.BuildApplicationSummaryViewModelWithModerationDetails(application,
+                            User.GetEmail());
+                    return View("~/Views/Roatp/ApplicationUnsuccessfulPostGateway.cshtml", unsuccessfulModel);
+
+                case ApplicationStatus.FeedbackAdded:
+                    return View("~/Views/Roatp/FeedbackAdded.cshtml", model);
+                case ApplicationStatus.Withdrawn:
+                    return View("~/Views/Roatp/ApplicationWithdrawn.cshtml", model);
+                case ApplicationStatus.Removed:
+                    return View("~/Views/Roatp/ApplicationWithdrawnESFA.cshtml", model);
```

Comment:
might want to rename this view to ApplicationRemoved.cshtml

Reply:
this is named like this to avoid usingthe words 'Removed' in the url

Reply:
URL will be 'ProcessApplicationStatus' for all of them